support-frontend/app/admin/settings/Amounts.scala

```diff
@@ -1,40 +1,52 @@
 package admin.settings
 
-import io.circe.generic.extras.Configuration
-import io.circe.generic.extras.auto._
-import io.circe.generic.extras.semiauto._
-import io.circe.{Decoder, Encoder}
+import com.gu.support.encoding.Codec.deriveCodec
+import com.gu.support.encoding.{Codec, DiscriminatedType}
 
 case class AmountsSelection(
     amounts: List[Int],
     defaultAmount: Int,
     hideChooseYourAmount: Option[Boolean],
 )
 
+object AmountsSelection {
+  implicit val codec: Codec[AmountsSelection] = deriveCodec[AmountsSelection]
```

Comment:
if you prefer you can reduce the repetition https://circe.github.io/circe/codecs/semiauto-derivation.html#:~:text=You%20can%20also,Foo%5D%20%3D%20deriveEncoder
```suggestion
  implicit val codec: Codec[AmountsSelection] = deriveCodec
```